tests/test_main.py

```diff
@@ -1 +1,51 @@
-# write your code here
+import pytest
+from unittest import mock
+
+
+from app.main import cryptocurrency_action
+
+
+@pytest.fixture()
+def mocked_get_exchange_rate_prediction():
+    with mock.patch('app.main.get_exchange_rate_prediction') as mocked:
+        yield mocked
+
+
+@pytest.mark.parametrize(
+    'current_rate, exchange_rate_prediction, awaited_result',
+    [
+        pytest.param(
+            1.0,
+            1.1,
+            "Buy more cryptocurrency",
+            id="Test when rate is more than +0.05"
+        ),
+        pytest.param(
+            1.0,
+            0.9,
+            'Sell all your cryptocurrency',
+            id="Test when next rate lower than -0.05"
+        ),
+        pytest.param(
+            1.0,
+            1.01,
+            'Do nothing',
+            id='Test when next rate in between 0 and +0.05'
+        ),
+        pytest.param(
+            1.0,
+            0.97,
+            "Do nothing",
+            id='Test when next rate in between -0.05 and 0'
+        )
+    ],
+)
```

Comment:
Add boundary conditions, you don't check when `get_exchange_rate_prediction` return value equals exactly 1.05 or 0.95 * value passed in `cryptocurrency_action`.